### PR TITLE
Update release instructions for Nengo Bones

### PR DESCRIPTION
**Pull request by tbekolay**
The process is a fair bit different now. Hopefully easier.

Still to do:

- [x] Finish the 0.2.0 Nengo Bones release using these instructions to make sure it's all accurate
- [x] Clarify if RC branches should be deleted

### CODE REVIEW

File: releasing.rst [hunk 1]
```diff
@@ -29,124 +27,89 @@ we do a few things to prepare:
...
-
-      python setup.py checkdocs
+8. Set the release date in the changelog, ``CHANGES.rst``.
 
-4. ``git add`` the changes above and make a release commit with
+9. ``git add`` the changes above and make a release commit with
 
    .. code-block:: bash
 
       git commit -m "Release version X.Y.Z"
```
Comment (drasmuss) [inline releasing.rst]: This is super minor, but in NengoDL I started doing "Release vX.Y.Z", just to line up with the tag and the next "Starting development of vX.Y.Z+1" commit.  Definitely doesn't matter, just occurred to me as I was reading this.

File: releasing.rst [hunk 2]
```diff
@@ -29,124 +27,89 @@ we do a few things to prepare:
...
+13. If any non-trivial changes were made in this process,
+    open a pull request on the ``release-candidate-X.Y.Z`` branch
+    to have the changes reviewed.
+
+Stage 2: Tagged release
+=======================
 
-      twine upload dist/<package-version>.tar.gz
-      twine upload dist/<package-version-extratags>.whl
+1. Merge the ``release-candidate-X.Y.Z`` branch into ``master``.
```
Comment (drasmuss) [inline releasing.rst]: I often do all this part in the `release-candidate` branch (and then merge at the end), just on the off chance that there's a mistake in e.g. step 3 or 4 which you'd want to force push to fix.

File: releasing.rst (hunk 1 above)
Comment (tbekolay) [reply releasing.rst]: Yeah I've been annoyed by the inconsistency before too, but erred on the side of consistency within the commit log. But yeah I think I'm in favor of switching so I'll change that.

File: releasing.rst (hunk 2 above)
Comment (tbekolay) [reply releasing.rst]: Makes sense, will move the merge to the end!

**Pull request closed without merging**